### DISCUSSION TITLE
Multi objectives not working as expected

### DISCUSSION THREAD

**Discussion by DeLaRiva** [optuna/optuna#3844] [Q&A]
Hi,

I'm trying to use Optuna for multi objectives: Maximize dose and minimize filament. I've to set around half a dozen parameters and the achieved dose will be measured. Thus, filament is actually a parameter Optuna has to choose from and to minimize simultaneously.
My code looks like

```python
def objective(trial)

    MIN_FIL = 0
    MAX_FIL = 1000000
    fil = trial.suggest_float("filament", MIN_FIL, MAX_FIL, step=100, log = False)

    MIN_VOL = 5000
    MAX_VOL = 200000
    vol = trial.suggest_float("filament", MIN_VOL , MAX_VOL , step=10, log = False)

    MIN_T = 0
    MAX_T = 10
    timing= trial.suggest_float("filament", MIN_T , MAX_T , step=0.1, log = False)
    (...)

    print("   Filament: ", {fil})
    (...)

    dose = float(input("Please enter the yielded dose: "))
    filament = fil

    return dose, filament
```

```python
       study = optuna.create_study(
                                    directions = ["maximize", "minimize"]
                                    )

        study.optimize(
                        lambda trial: objective(trial)
                        , n_trials = n_trials
                        , show_progress_bar=True
                        )
```

But as far as I can tell the algorithm didn't really try to minimize the filament. I checked only for three experiments but it didn't work as expected. Maybe I have to study further runs but my gut feeling says something is not correct the way I am doing it.
Here is the pareto plot.

![497_pareto_run1_1](https://user-images.githubusercontent.com/29283287/182816415-6b3439db-ec49-4ea4-a089-851685c85678.png)

As for the parameter importance plot of the filament one can see that the filament is important as ~100 %. As expected(?)

Does it make sense to feed back the filament immediately after letting it to be sampled?

**Comment by nzw0301** [comment]
I'm not sure how to compute `Dose`, but the axis shows too huge a range (0 -- 1.4M), so as you said adding more trials might be good approach or rescale the xaxis scale for visualisation.

**Comment by DeLaRiva** [comment]
Thank you! As far as I can [see ](https://optuna.readthedocs.io/en/stable/reference/visualization/generated/optuna.visualization.plot_optimization_history.html#optuna.visualization.plot_optimization_history)there is no option to rescale?
The dose is actually not calculated but yielded by external devices. That's why I put in the values for it manually through the terminal. A technical implementation is quite an effort so I've to check it's working and safe and sound.
The parameters resp. the found optima by Optuna will be used to control x-ray devices later so it's important it is working.

Despite from that I didn't make any (obvious) mistake when using the multi objectives?
I'm especially a bit concerned about redirecting the filament immediately after Optuna sampled from it right before(?).
